posthog/temporal/common/posthog_client.py

```diff
@@ -12,6 +13,19 @@
     WorkflowInterceptorClassInput,
 )
 
+logger = structlog.get_logger()
+
+
+async def _add_inputs_to_properties(properties: dict[str, Any], input: ExecuteActivityInput | ExecuteWorkflowInput):
+    if len(input.args) == 1 and is_dataclass(input.args[0]) and hasattr(input.args[0], "properties_to_log"):
```

Comment:
I think I would rather have the class just return a dict of properties based on its `properties_to_log`. It feels somewhat disconnected to have this in a separate module.

Reply:
Yeah could do. I chose to add the logic here just to remove a bit of redundancy in the dataclasses. For example, currently you need to do something like the following in your dataclass:

```python
    def properties_to_log(self) -> list[str]:
        return ["team_id", "job_id", "schema_id", "source_id", "status"]
```

whereas if you made it return a dict you would have to do:

```python
    def properties_to_log(self) -> dict[str, Any]:
        return {
            "team_id": self.team_id,
            "job_id": self.job_id,
            "schema_id": self.schema_id,
            "source_id": self.source_id,
            "status": self.status,
        }
```

I suppose it's not too much work though, and would actually get some type annotation benefits from doing it this way